build.cmd

```diff
@@ -189,429 +148,256 @@ set "__TestRootDir=%__RootBinDir%\tests"
-  exit /b 1
-)
+
+@call %~dp0run.cmd build -generateHeaderWindows -NativeVersionHeaderFile="%__RootBinDir%\obj\_version.h" %__runArgs% %__unprocessedBuildArgs% 
 
 REM =========================================================================================
 REM ===
-REM === Start the build steps
+REM === Build the CLR VM
 REM ===
 REM =========================================================================================
 
-:: Generate _version.h
-if exist "%__RootBinDir%\obj\_version.h" del "%__RootBinDir%\obj\_version.h"
-%_msbuildexe% "%__ProjectFilesDir%\build.proj" /t:GenerateVersionHeader /v:minimal /p:NativeVersionHeaderFile="%__RootBinDir%\obj\_version.h" /p:GenerateVersionHeader=true %__OfficialBuildIdArg%
-if defined __CoreLibOnly goto PerformCoreLibBuild
+if %_buildNative% EQU 1 ( 
+    echo %__MsgPrefix%Commencing build of native components for %__BuildOS%.%__BuildArch%.%__BuildType%
 
-if defined __SkipNativeBuild (
-    echo %__MsgPrefix%Skipping native components build
-    goto SkipNativeBuild
-)
+	set nativePlatfromArgs=-platform=%__BuildArch%
+    if /i "%__BuildArch%" == "arm64" ( set nativePlatfromArgs=-useEnv )
 
-echo %__MsgPrefix%Commencing build of native components for %__BuildOS%.%__BuildArch%.%__BuildType%
+    set __msbuildLog=/flp:Verbosity=normal;LogFile="%__LogsDir%\CoreCLR_%__BuildOS%__%__BuildArch%__%__BuildType%.log"
+    set __msbuildWrn=/flp1:WarningsOnly;LogFile="%__LogsDir%\CoreCLR_%__BuildOS%__%__BuildArch%__%__BuildType%.wrn"
+    set __msbuildErr=/flp2:ErrorsOnly;LogFile="%__LogsDir%\CoreCLR_%__BuildOS%__%__BuildArch%__%__BuildType%.err"
 
-REM Use setlocal to restrict environment changes form vcvarsall.bat and more to just this native components build section.
-setlocal EnableDelayedExpansion EnableExtensions
-
-if /i "%__BuildArch%" == "arm64" ( 
-rem arm64 builds currently use private toolset which has not been released yet
-REM TODO, remove once the toolset is open.
-call :PrivateToolSet
-
-goto GenVSSolution
-)
-
-:: Set the environment for the native build
-set __VCBuildArch=x86_amd64
-if /i "%__BuildArch%" == "x86" (set __VCBuildArch=x86)
-echo %__MsgPrefix%Using environment: "%__VSToolsRoot%\..\..\VC\vcvarsall.bat" %__VCBuildArch%
-call                                 "%__VSToolsRoot%\..\..\VC\vcvarsall.bat" %__VCBuildArch%
-@if defined __echo @echo on
+    if /i "%__BuildArch%" == "arm64" ( 
+        rem arm64 builds currently use private toolset which has not been released yet
+        REM TODO, remove once the toolset is open.
+        call :PrivateToolSet
+        goto GenVSSolution
+    )
 
-if not defined VSINSTALLDIR (
-    echo %__MsgPrefix%Error: VSINSTALLDIR variable not defined.
-    exit /b 1
-)
-if not exist "%VSINSTALLDIR%DIA SDK" goto NoDIA
+    :: Set the environment for the native build
+    set __VCBuildArch=x86_amd64
+    if /i "%__BuildArch%" == "x86" ( set __VCBuildArch=x86 )
+    echo %__MsgPrefix%Using environment: "%VS140COMNTOOLS%\..\..\VC\vcvarsall.bat" %__VCBuildArch%
```

Comment:
Why have you replaced the __VSToolsRoot with VS140COMNTOOLS everywhere? Now when we will change the version of tools we use in the future, it will need to be modified at 17 different places instead of just one.